src/parse.jl

```diff
@@ -0,0 +1,254 @@
+"""
+    count_locations_and_transitions(root_sxmodel)
+
+Returns the number of locations and transitions for each component.
+
+### Input
+
+- `root_sxmodel` -- the root element of a SX file
+
+### Output
+
+Two vectors of integers `(lcount, tcount)`, where the i-th entry of `lcount` and
+`tcount` are number of locations and transitions, respectively, of the i-th component.
+"""
+function count_locations_and_transitions(root_sxmodel)
+    lcount = Vector{Int}() # num locations for each component
+    tcount = Vector{Int}() # num transitions for each component
+    id_component = 0 # index of the current component
+
+    # count the number of locations for each component
+    for component in eachelement(root_sxmodel)
+        if nodename(component) == "component"
+            id_component += 1
+            push!(lcount, 0)
+            push!(tcount, 0)
+            for field in eachelement(component)
+                if nodename(field) == "location"
+                    lcount[id_component] += 1
+                elseif nodename(field) == "transition"
+                    tcount[id_component] += 1
+                end
```

Comment:
I would print a warning in the missing `else` case.

Reply:
this function only wants to count locations and transitions, i wouldn't print a warning for other things they should be just not taken into account in the count.